appverifier/src/main/java/com/android/mdl/appreader/readercertgen/CertificateGenerator.kt

```diff
@@ -0,0 +1,108 @@
+package com.android.mdl.appreader.readercertgen
+
+import org.bouncycastle.asn1.ASN1ObjectIdentifier
+import org.bouncycastle.asn1.x500.X500Name
+import org.bouncycastle.asn1.x509.BasicConstraints
+import org.bouncycastle.asn1.x509.ExtendedKeyUsage
+import org.bouncycastle.asn1.x509.Extension
+import org.bouncycastle.asn1.x509.GeneralName
+import org.bouncycastle.asn1.x509.GeneralNames
+import org.bouncycastle.asn1.x509.KeyPurposeId
+import org.bouncycastle.asn1.x509.KeyUsage
+import org.bouncycastle.asn1.x509.SubjectKeyIdentifier
+import org.bouncycastle.cert.CertIOException
+import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter
+import org.bouncycastle.cert.jcajce.JcaX509ExtensionUtils
+import org.bouncycastle.cert.jcajce.JcaX509v3CertificateBuilder
+import org.bouncycastle.operator.OperatorCreationException
+import org.bouncycastle.operator.jcajce.JcaContentSignerBuilder
+import java.io.IOException
+import java.security.NoSuchAlgorithmException
+import java.security.cert.CertificateException
+import java.security.cert.X509Certificate
+import java.util.Optional
+
+object CertificateGenerator {
+    private const val CRITICAL = true
+    private const val NOT_CRITICAL = false
+    @JvmStatic
+    @Throws(CertIOException::class, CertificateException::class, OperatorCreationException::class)
+    fun generateCertificate(
+        data: DataMaterial,
+        certMaterial: CertificateMaterial,
+        keyMaterial: KeyMaterial
+    ): X509Certificate {
+        val issuerCert: Optional<X509Certificate> = keyMaterial.issuerCertificate
+        val subjectDN = X500Name(data.subjectDN)
+        // doesn't work, get's reordered
+        // issuerCert.isPresent() ? new X500Name(issuerCert.get().getSubjectX500Principal().getName()) : subjectDN;
+        val issuerDN = X500Name(data.issuerDN)
+        val contentSigner =
+            JcaContentSignerBuilder(keyMaterial.signingAlgorithm).build(keyMaterial.signingKey)
+        val certBuilder = JcaX509v3CertificateBuilder(
+            issuerDN,
+            certMaterial.serialNumber,
+            certMaterial.startDate, certMaterial.endDate,
+            subjectDN,
+            keyMaterial.publicKey
+        )
+
+
+        // Extensions --------------------------
+        val jcaX509ExtensionUtils: JcaX509ExtensionUtils
+        jcaX509ExtensionUtils = try {
```

Comment:
This can be defined into a single expression, like so:

```
val jcaX509ExtensionUtils = try { 
  ... 
} catch { 
  ... 
}
```
or even

```
the type will be inferred automatically